components/menu/menu_item.tsx

```diff
@@ -0,0 +1,208 @@
+ * Checkout Compass's Menu Item(compass.mattermost.com)  for terminology, styling and usage guidelines.
+ *
+ * @example
+ * <Menu.Container>
+ *    <Menu.Item/>
+ * </Menu.Container>
+ */
+export function MenuItem(props: Props) {
+    const {
+        leadingElement,
+        labels,
+        trailingElements,
+        isDestructive,
+        children,
+        ...restProps
+    } = props;
+
+    // When both primary and secondary labels are passed, we need to apply minor changes to the styling. Check below in styled component for more details.
+    const hasSecondaryLabel = labels && labels.props && labels.props.children && Children.count(labels.props.children) === 2;
+
+    return (
+        <MenuItemStyled
+            disableRipple={true}
+            disableTouchRipple={true}
+            isDestructive={isDestructive}
+            hasSecondaryLabel={hasSecondaryLabel}
+            {...restProps}
+        >
+            {leadingElement && <div className='leading-element'>{leadingElement}</div>}
+            <div className='label-elements'>{labels}</div>
+            {trailingElements && <div className='trailing-elements'>{trailingElements}</div>}
+            {children}
+        </MenuItemStyled>
+    );
+}
+
+interface MenuItemStyledProps extends MuiMenuItemProps {
+    isDestructive?: boolean;
+    hasSecondaryLabel?: boolean;
+}
+
+/* eslint-disable no-negated-condition */
+const MenuItemStyled = styled(MuiMenuItem, {
+    shouldForwardProp: (prop) => prop !== 'isDestructive' && prop !== 'hasSecondaryLabel',
+})<MenuItemStyledProps>(
+    ({isDestructive = false, hasSecondaryLabel = false}) => ({
+        '&.MuiMenuItem-root': {
+            fontFamily: '"Open Sans", sans-serif',
+            color: !isDestructive ? 'var(--center-channel-color)' : 'var(--error-text)',
+            padding: '6px 20px',
+            display: 'flex',
+            flexDirection: 'row',
+            flexWrap: 'nowrap',
+            justifyContent: 'flex-start',
+            alignItems: !hasSecondaryLabel ? 'center' : 'flex-start',
+            minHeight: '36px',
+            maxHeight: '56px',
+
+            // aria expanded to add the active styling on parent sub menu item
+            '&.Mui-active, &[aria-expanded="true"]': {
+                'background-color': !isDestructive ? 'rgba(var(--button-bg-rgb), 0.08)' : 'background-color: rgba(var(--error-text-color-rgb), 0.16)',
+            },
+
+            '&:hover': {
+                backgroundColor: !isDestructive ? 'rgba(var(--center-channel-color-rgb), 0.08)' : 'var(--error-text)',
+                color: isDestructive && 'var(--button-color)',
+            },
+
+            '&.Mui-disabled': {
+                color: 'rgba(var(--center-channel-color-rgb), 0.32)',
+            },
+
+            '&.Mui-focusVisible': {
+                boxShadow: !isDestructive ? '0 0 0 2px var(--denim-sidebar-active-border) inset' : '0 0 0 2px rgba(var(--button-color-rgb), 0.16) inset',
```

Comment:
@M-ZubairAhmed, I missed it earlier, this should be 
```suggestion
                boxShadow: !isDestructive ? '0 0 0 2px var(--sidebar-text-active-border) inset' : '0 0 0 2px rgba(var(--button-color-rgb), 0.16) inset',
```

Reply:
changed